cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -620,7 +612,8 @@ protected PersistentVolumeClaim createPersistentVolumeClaim(int podNumber, Strin
                 .withNewMetadata()
                     .withName(name)
                     .withNamespace(namespace)
-                    .withLabels(mergeLabelsOrAnnotations(getLabelsWithName(templateStatefulSetLabels), templatePersistentVolumeClaimLabels))
+                    // labels with the Strimzi name set to that of the component
+                    .withLabels(getLabelsWithStrimziName(this.name, templatePersistentVolumeClaimLabels).toMap())
```

Comment:
BEfore we included here also the `templateStatefulSetLabels`. I'm sure that had some reason. Why don't we need it anymore?

Reply:
We do need it, but I have moved the merge to arguably the correct place to put it, in each model when the labels are initially defined, I think merging the labels this late is an anti-pattern and causes more confusion.
https://github.com/strimzi/strimzi-kafka-operator/pull/2730/files#diff-405b3358597333915e4058062355209aL656
https://github.com/strimzi/strimzi-kafka-operator/pull/2730/files#diff-93b151183726e7a1a3e42dcffc82367bR320-R321

Reply:
Ok, great. I didn't noticed that.